met-web/src/components/comments/admin/reviewListing/index.tsx

```diff
@@ -38,35 +42,48 @@ const CommentListing = () => {
 
     const { page, size, sort_key, nested_sort_key, sort_order } = paginationOptions;
 
-    const loadComments = async () => {
+    const loadSurvey = async () => {
         try {
-            if (isNaN(Number(surveyId))) {
-                dispatch(openNotification({ severity: 'error', text: 'Invalid surveyId' }));
-            }
+            const survey = await getSurvey(Number(surveyId));
+            setSurvey(survey);
+        } catch (error) {
+            console.log(error);
+            dispatch(openNotification({ severity: 'error', text: 'Error occurred while fetching survey information' }));
+            setTableLoading(false);
+        }
+    };
 
+    const loadSubmissions = async () => {
+        try {
             setTableLoading(true);
-            const response = await getCommentsPage({
+            const response = await getSubmissionPage({
                 survey_id: Number(surveyId),
                 page,
                 size,
                 sort_key: nested_sort_key || sort_key,
                 sort_order,
                 search_text: searchFilter.value,
             });
-            setComments(response.items);
+            setSubmissions(response.items);
             setPageInfo({
                 total: response.total,
             });
             setTableLoading(false);
```

Comment:
I guess this tableLoading should be set to false when we know loadSurvey and loadSubmissions are both done